### PR TITLE
Add search for WPS services in Csw Catalog Groups

### PR DESCRIPTION
**Pull request by sppigot**
Also add findIndex to avoid breaking IE11

### CODE REVIEW

File: lib/Models/CswCatalogGroup.js [hunk 1]
```diff
@@ -252,7 +261,13 @@ var resourceFormats = [
 ];
 
 CswCatalogGroup.prototype._load = function() {
-    var postDataTemplate = defaultValue(this.getRecordsTemplate, CswCatalogGroup.defaultGetRecordsTemplate);
+    var postDataTemplate;
+
+    if (this.includeWps) {
```
Comment (kring) [inline lib/Models/CswCatalogGroup.js]: The way this is set up, if `includeWps` is true, any WMS items are not included, regardless of the state of `includeWms`.

File: lib/Models/CswCatalogGroup.js [hunk 2]
```diff
@@ -447,13 +481,29 @@ Otherwise, if reloading doesn\'t fix it, please report the problem by sending an
...
+    wps.name = title;
+    wps.url = uri;
+
+    return wps.load().then(function() {
+        return wps;
+    });
+}
+
+function addProcessesToGroup(wpsPromise, wpsGroup) {
+    wpsPromise.then(function(wpsItem) {
```
Comment (kring) [inline lib/Models/CswCatalogGroup.js]: Without an `otherwise`, a WPS load error will silently fail telling the user about it.

Comment (kring) [inline lib/Models/CswCatalogGroup.js]: The best way to handle this case is to make sure that all promises are returned from `_load`.  Then other parts of TerriaJS will automatically handle errors and display them to the user.

File: lib/Models/CswCatalogGroup.js [hunk 3]
```diff
@@ -599,4 +649,37 @@ function cleanUrl(url) {
     return uri.toString();
 }
 
+function cleanAndProxyUrl(catalogGroup, url) {
+    return proxyCatalogItemUrl(catalogGroup, cleanUrl(url), '1d');
+}
+
+function proxyUrl(terria, url) {
```
Comment (kring) [inline lib/Models/CswCatalogGroup.js]: This function isn't used.

File: lib/Models/CswCatalogGroup.js [hunk 4]
```diff
@@ -599,4 +649,37 @@ function cleanUrl(url) {
...
+    if (defined(terria.corsProxy) && terria.corsProxy.shouldUseProxy(url)) {
+          return terria.corsProxy.getURL(url);
+    }
+    return url;
+}
+
+// polyFill from ecma6 js
+function findIndex(array, predicate) {
```
Comment (kring) [inline lib/Models/CswCatalogGroup.js]: It would be good to make this a separate module in `Core`.

**Comment by kring** [comment]
Thanks @sppigot!  Just those few comments.

**Comment by sppigot** [comment]
Thanks @kring! - removal of (old) special case handling for wps stuff should deal with all the comments.

**Comment by kring** [comment]
Thanks @sppigot!  I'm going to merge this via #1525.

**Pull request merged**